### ISSUE TITLE
Error on `buildah run` (version 0.8)

### ISSUE THREAD

**Issue by bertinatto** [projectatomic/buildah#334]
Hi,

I am getting an error when I execute `buildah run`. I hadn't seen this in happening in 0.7.

Here is how I reproduce it:

```
➜  buildah (git)-[master] ✗ make && sudo ./buildah from registry.access.redhat.com/rhscl/redis-32-rhel7
make -C docs
make[1]: Entering directory '/home/fjb/src/github.com/projectatomic/buildah/docs'
make[1]: Nothing to be done for 'docs'.
make[1]: Leaving directory '/home/fjb/src/github.com/projectatomic/buildah/docs'
redis-32-rhel7-working-container-6

➜  buildah (git)-[master] ✗ sudo ./buildah --debug run redis-32-rhel7-working-container-6 sh
DEBU[0000] [graphdriver] trying provided driver "overlay"
DEBU[0000] overlay: override_kernelcheck=true
DEBU[0000] backingFs=extfs,  projectQuotaSupported=false
DEBU[0000] using "/tmp/buildah502786698" to hold bundle data
WARN[0000] file "/etc/containers/mounts.conf" not found, skipping...
WARN[0000] file "/usr/share/containers/mounts.conf" not found, skipping...
DEBU[0000] CopyFileWithTar(/var/lib/containers/storage/overlay/e820a7a861106a0a326c56cc881feeb323427912f3d1b7e03a04f4461002e428/merged/var/lib/redis/data, /var/lib/containers/storage/overlay-containers/b44b3e2d9f6580a282b8aa4fd01f212affc4222c611c5d4e4db84ea74992b914/userdata/buildah-volumes/e288f1674d08bece623eefaccadd92b88fef68548b2e13aad1096401f2da6a38)
DEBU[0000] error running [sh] in container "redis-32-rhel7-working-container-6": error resolving mountpoints for container: error populating directory "/var/lib/containers/storage/overlay-containers/b44b3e2d9f6580a282b8aa4fd01f212affc4222c611c5d4e4db84ea74992b914/userdata/buildah-volumes/e288f1674d08bece623eefaccadd92b88fef68548b2e13aad1096401f2da6a38" for volume "/var/lib/redis/data" in container "b44b3e2d9f6580a282b8aa4fd01f212affc4222c611c5d4e4db84ea74992b914" using contents of "/var/lib/containers/storage/overlay/e820a7a861106a0a326c56cc881feeb323427912f3d1b7e03a04f4461002e428/merged/var/lib/redis/data": Can't copy a directory
ERRO[0000] error resolving mountpoints for container: error populating directory "/var/lib/containers/storage/overlay-containers/b44b3e2d9f6580a282b8aa4fd01f212affc4222c611c5d4e4db84ea74992b914/userdata/buildah-volumes/e288f1674d08bece623eefaccadd92b88fef68548b2e13aad1096401f2da6a38" for volume "/var/lib/redis/data" in container "b44b3e2d9f6580a282b8aa4fd01f212affc4222c611c5d4e4db84ea74992b914" using contents of "/var/lib/containers/storage/overlay/e820a7a861106a0a326c56cc881feeb323427912f3d1b7e03a04f4461002e428/merged/var/lib/redis/data": Can't copy a directory

➜  buildah (git)-[master] ✗ sudo ./buildah --debug run redis-32-rhel7-working-container-6 sh
DEBU[0000] [graphdriver] trying provided driver "overlay"
DEBU[0000] overlay: override_kernelcheck=true
DEBU[0000] backingFs=extfs,  projectQuotaSupported=false
DEBU[0000] using "/tmp/buildah722274281" to hold bundle data
WARN[0000] file "/etc/containers/mounts.conf" not found, skipping...
WARN[0000] file "/usr/share/containers/mounts.conf" not found, skipping...
DEBU[0000] config = {"ociVersion":"1.0.0","process":{"terminal":true,"user":{"uid":1001,"gid":998},"args":["sh"],"env":["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin","TERM=xterm","container=

(...)

```

As we can see, I was able to `run` my command in the second attempt.

Thanks!

**Comment by TomSweeneyRedHat** [comment]
@bertinatto Thanks for the problem report.  Just to make sure I have this right, you've run "buildah run" and it failed then you did the same "buildah run" and it worked?

Are you able to do 'journalctl | less' and see any related errors or avc's?  If so could you add them to this issue?

Does the fail then work strategy hold for all containers?

Finally, which operating system were you running on?  Had you installed a kit or cloned from git and built your own?  @rhatdan has been working through some bug/selinux related bugs with buildah run this morning, the fix may be in place already there if you can refresh your build environment.

**Comment by bertinatto** [comment]
Thanks for replying @TomSweeneyRedHat.

> Just to make sure I have this right, you've run "buildah run" and it failed then you did the same "buildah run" and it worked?

That's exactly right.

> Does the fail then work strategy hold for all containers?

Not sure if I understand correctly, but the problem still happens if I create another container. The `run` command only succeeds in the second attempt. 

> Finally, which operating system were you running on?

Fedora 26.

> Had you installed a kit or cloned from git and built your own?

I'm currently using tip:

```
commit bf40000e72b351067ebae7b77d212a200f9ce051
Author: Daniel J Walsh <dwalsh@redhat.com>
Date:   Wed Nov 22 16:35:41 2017 +0000

    Bump to v0.8
```

> Are you able to do 'journalctl | less' and see any related errors or avc's? If so could you add them to this issue?

Of course, sorry for missing this.

I just reproduced the problem following the same steps:

```
➜  buildah (git)-[master] ✗ make && sudo ./buildah from registry.access.redhat.com/rhscl/redis-32-rhel7
go build -ldflags '-X main.gitCommit=bf40000 -X main.buildInfo=1511421147' -o buildah -tags "   selinux seccomp" ./cmd/buildah
go build -ldflags '-X main.gitCommit=bf40000 -X main.buildInfo=1511421147' -o imgtype -tags "   selinux seccomp" ./tests/imgtype.go
make -C docs
make[1]: Entering directory '/home/fjb/src/github.com/projectatomic/buildah/docs'
make[1]: Nothing to be done for 'docs'.
make[1]: Leaving directory '/home/fjb/src/github.com/projectatomic/buildah/docs'
[sudo] password for fjb:
redis-32-rhel7-working-container-7

➜  buildah (git)-[master] ✗ sudo ./buildah --debug run redis-32-rhel7-working-container-7 sh
DEBU[0000] [graphdriver] trying provided driver "overlay"
DEBU[0000] overlay: override_kernelcheck=true
DEBU[0000] backingFs=extfs,  projectQuotaSupported=false
DEBU[0000] using "/tmp/buildah109154154" to hold bundle data
WARN[0000] file "/etc/containers/mounts.conf" not found, skipping...
WARN[0000] file "/usr/share/containers/mounts.conf" not found, skipping...
DEBU[0000] CopyFileWithTar(/var/lib/containers/storage/overlay/c40a0eb193b81bf89ed125027a5c473cdce2351c7edcb797d3c6458c9b33ab1e/merged/var/lib/redis/data, /var/lib/containers/storage/overlay-containers/142193a1871d8496a9becdfda3d435b93b98e605792068f2fb7619579953a1a7/userdata/buildah-volumes/e288f1674d08bece623eefaccadd92b88fef68548b2e13aad1096401f2da6a38)
DEBU[0000] error running [sh] in container "redis-32-rhel7-working-container-7": error resolving mountpoints for container: error populating directory "/var/lib/containers/storage/overlay-containers/142193a1871d8496a9becdfda3d435b93b98e605792068f2fb7619579953a1a7/userdata/buildah-volumes/e288f1674d08bece623eefaccadd92b88fef68548b2e13aad1096401f2da6a38" for volume "/var/lib/redis/data" in container "142193a1871d8496a9becdfda3d435b93b98e605792068f2fb7619579953a1a7" using contents of "/var/lib/containers/storage/overlay/c40a0eb193b81bf89ed125027a5c473cdce2351c7edcb797d3c6458c9b33ab1e/merged/var/lib/redis/data": Can't copy a directory
ERRO[0000] error resolving mountpoints for container: error populating directory "/var/lib/containers/storage/overlay-containers/142193a1871d8496a9becdfda3d435b93b98e605792068f2fb7619579953a1a7/userdata/buildah-volumes/e288f1674d08bece623eefaccadd92b88fef68548b2e13aad1096401f2da6a38" for volume "/var/lib/redis/data" in container "142193a1871d8496a9becdfda3d435b93b98e605792068f2fb7619579953a1a7" using contents of "/var/lib/containers/storage/overlay/c40a0eb193b81bf89ed125027a5c473cdce2351c7edcb797d3c6458c9b33ab1e/merged/var/lib/redis/data": Can't copy a directory

➜  buildah (git)-[master] ✗ sudo ./buildah --debug run redis-32-rhel7-working-container-7 sh
DEBU[0000] [graphdriver] trying provided driver "overlay"
(...)

$
```

And this time I collected the logs with `journalctl -f`: https://gist.github.com/bertinatto/6b281d3e7aed32d696499271bf4d7483

Please, let me know if I can provide you with any other information.

Thanks!

**Comment by rhatdan** [comment]
Are you sure the first one is not working, or you are not getting a # prompt.

Try to execute buildah run ls /

**Comment by bertinatto** [comment]
I created a new container and  tried to run `ls /` in it. Got the same results:

```
➜  buildah (git)-[master] ✗ sudo ./buildah --debug run redis-32-rhel7-working-container-9 ls /
DEBU[0000] [graphdriver] trying provided driver "overlay"
DEBU[0000] overlay: override_kernelcheck=true
DEBU[0000] backingFs=extfs,  projectQuotaSupported=false
DEBU[0000] using "/tmp/buildah214186068" to hold bundle data
WARN[0000] file "/etc/containers/mounts.conf" not found, skipping...
WARN[0000] file "/usr/share/containers/mounts.conf" not found, skipping...
DEBU[0000] CopyFileWithTar(/var/lib/containers/storage/overlay/9e5f622b3625b489f1d931c6ccf0beaaa9338ecff6a9ec813ca22d4107fdda0c/merged/var/lib/redis/data, /var/lib/containers/storage/overlay-containers/af575a211a6c55e2af2baa561e26f818d37afd0d18621c4e007cb62e781c5f9f/userdata/buildah-volumes/e288f1674d08bece623eefaccadd92b88fef68548b2e13aad1096401f2da6a38)
DEBU[0000] error running [ls /] in container "redis-32-rhel7-working-container-9": error resolving mountpoints for container: error populating directory "/var/lib/containers/storage/overlay-containers/af575a211a6c55e2af2baa561e26f818d37afd0d18621c4e007cb62e781c5f9f/userdata/buildah-volumes/e288f1674d08bece623eefaccadd92b88fef68548b2e13aad1096401f2da6a38" for volume "/var/lib/redis/data" in container "af575a211a6c55e2af2baa561e26f818d37afd0d18621c4e007cb62e781c5f9f" using contents of "/var/lib/containers/storage/overlay/9e5f622b3625b489f1d931c6ccf0beaaa9338ecff6a9ec813ca22d4107fdda0c/merged/var/lib/redis/data": Can't copy a directory
ERRO[0000] error resolving mountpoints for container: error populating directory "/var/lib/containers/storage/overlay-containers/af575a211a6c55e2af2baa561e26f818d37afd0d18621c4e007cb62e781c5f9f/userdata/buildah-volumes/e288f1674d08bece623eefaccadd92b88fef68548b2e13aad1096401f2da6a38" for volume "/var/lib/redis/data" in container "af575a211a6c55e2af2baa561e26f818d37afd0d18621c4e007cb62e781c5f9f" using contents of "/var/lib/containers/storage/overlay/9e5f622b3625b489f1d931c6ccf0beaaa9338ecff6a9ec813ca22d4107fdda0c/merged/var/lib/redis/data": Can't copy a directory
```

Also, this error seems to be happening with this specific image. I just tried with a different image and it worked:

```
➜  buildah (git)-[master] ✗ sudo ./buildah from registry.access.redhat.com/rhel7/rhel-atomic
rhel-atomic-working-container-2

➜  buildah (git)-[master] ✗ sudo ./buildah --debug run rhel-atomic-working-container-2 ls /
(...)
bin   dev  home  lib64       media  opt   root  sbin  sys  usr
boot  etc  lib   lost+found  mnt    proc  run   srv   tmp  var
```

**Comment by TomSweeneyRedHat** [comment]
@rhatdan @nalind this appears to be failing in run.go trying to bind mount /etc/hosts and/or etc/resolv.conf.  Is there something funky about those in a redis container?

```
        bindFiles := []string{"/etc/hosts", "/etc/resolv.conf"}
        err = b.setupMounts(mountPoint, spec, options.Mounts, bindFiles, b.Volumes())
        if err != nil {
                return errors.Wrapf(err, "error resolving mountpoints for container")
        }
```

**Comment by TomSweeneyRedHat** [comment]
Closing as this issue was fixed in #476 and will be in version 0.12.
